### DISCUSSION TITLE
Should "pointerover" events fire when a game object is not visible?

### DISCUSSION THREAD

**Discussion by scottwestover** [photonstorm/phaser#6389] [Q&A] [answered]
I ran into a an issue where the `pointerover` event is not firing for a game object after I set the alpha for that game object to `0`. I was not sure if this is expected behavior, or if this is a bug.

I am seeing this behavior in Phaser v3.55.2.

Here is an example code snippet that shows the functionality I am trying to achieve.
```javascript
const rect = this.add
  .rectangle(50, 50, 100, 100, 0xffff00)
  .setInteractive()
  .setAlpha(0.2);

rect.on("pointerover", () => {
  console.log("pointerover");
  rect.setAlpha(0.6);
});
rect.on("pointerout", () => {
  console.log("pointerout");
  rect.setAlpha(0);
});

const rect2 = this.add
  .rectangle(50, 200, 100, 100, 0xffff00)
  .setInteractive()
  .setAlpha(0.2);

rect2.on("pointerover", () => {
  rect2.setAlpha(0.6);
});
rect2.on("pointerout", () => {
  rect2.setAlpha(0.1);
});
```

In the code snippet above, and in the codepen link below, when you hover over the top rectangle, `rect`, the rectangle will show and then disappear as expected. However, once you hover over the rectangle a second time, the `pointerover` event does not fire again.

If you hover over the bottom rectangle, `rect2`, the rectangle will show and then disappear multiple times as you move the mouse back and forth over the game object.

Here is a live example showing the behavior I am seeing: https://codepen.io/swesto01/pen/mdGdQvj

I was able to work around the issue by setting my alpha value to a very low value like `0.01`, but wanted to ask here what the expected behavior should be.

**Comment by photonstorm** [comment, marked as answer]
Game Objects that don't render (i.e. visible false or alpha zero) are skipped for input, yes. This is how it was designed from the start so is the default standard behaviour. For non-rendering Game Objects that need input (i.e. for drop zones) there is the Zone Game Object.

**Reply by scottwestover** [reply]
Thanks for the quick response, and for confirming that is the intended behavior! I will take a look at the Zone Game Object.
